Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/format/xml/Space.java

```diff
@@ -0,0 +1,179 @@
+        /**
+         * Last character was text. If the next character is text, do not
+         * prepend a white space.
+         */
+        TEXT,
+        /**
+         * Last character was non-initial white space. If the next character is
+         * text, a white space will be prepended.
+         */
+        WHITE_SPACE
+    }
+
+    /**
+     * Reverse look-up map for the enum constants.
+     */
+    private static final Map<String, Space> reversed;
+
+    /**
+     * Characters considered as white space.
+     */
+    private static Set<Integer> WHITE_SPACE = new HashSet<>(
+            new HashSet<>(Arrays.asList((int) '\n', (int) '\r', (int) '\t', (int) ' ')));
+
+    static {
+        reversed = new HashMap<>();
+        for (final Space value : Space.values()) {
+            reversed.put(value.enumeratedValue, value);
+        }
+    }
+
+    /**
+     * Returns whether white space must be preserved to correctly represent this
+     * string in XML.
+     *
+     * @param s
+     *            string to examine
+     * @return if preservation of white space is necessary
+     */
+    public static boolean requiresPreservation(String s) {
+        boolean result = true;
+        for (int i = 0; i < s.length(); i++) {
+            int codePoint = s.codePointAt(i);
+            if (!WHITE_SPACE.contains(codePoint)) {
+                result = false;
+            } else if ((codePoint == ' ') && !result) {
+                result = true;
+            } else {
+                return true;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Resolve a String to an enum constant.
+     *
+     * @param value
+     *            String to resolve
+     * @return the enum constant
+     */
+    static Space resolve(String value) {
+        String lowerCase = value.toLowerCase();
+        if (!reversed.containsKey(lowerCase)) {
+            throw new IllegalArgumentException("Unknown xml:space value: " + value);
+        }
+        return reversed.get(lowerCase);
+    }
+
+    /**
+     * String value as by specification.
+     */
+    private final String enumeratedValue;
+
+    /**
+     * Creates the enumeration’s elements. Populates their private fields.
+     *
+     * @param enumeratedValue
+     *            String value as by specification
+     * @param toPreserve
+     *            whether the white-space is to preserve
+     */
+    private Space(String enumeratedValue) {
+        this.enumeratedValue = enumeratedValue;
+    }
+
+    /**
+     * Returns the string value representing this setting.
+     *
+     * @return string value representing this setting
+     */
+    public String getValue() {
+        return enumeratedValue;
+    }
+
+    /**
+     * Trims a String the XML way, if Space.DEFAULT is chosen. Does nothing if
+     * Space.PRESERVE is chosen.
+     *
+     * @param s
+     *            string to process
+     * @return trimmed string
+     */
+    public abstract String trim(String s);
```

Comment:
Short method parameter names should be avoided. Please use a longer and understandable name.